rotkehlchen/tasks/manager.py

```diff
@@ -538,6 +541,36 @@ def _maybe_update_snapshot_balances(self) -> Optional[list[gevent.Greenlet]]:
         """
         with self.database.conn.read_ctx() as cursor:
             if self.database.should_save_balances(cursor):
+                detect_tokens: dict[SupportedBlockchain, list[ChecksumEvmAddress]] = defaultdict(list)  # noqa: E501
+                if CachedSettings().get_settings().auto_detect_tokens is True:
+                    type_filter = [event_type.serialize() for event_type in [
+                        HistoryEventType.RECEIVE,
+                        HistoryEventType.WITHDRAWAL,
+                    ]]
+                    subtype_filter = [event_subtype.serialize() for event_subtype in [
+                        HistoryEventSubType.RECEIVE,
+                        HistoryEventSubType.REWARD,
+                        HistoryEventSubType.REMOVE_ASSET,
+                        HistoryEventSubType.MEV_REWARD,
+                    ]]
+                    last_save_time = self.database.get_last_balance_save_time(cursor)
+                    for chain in EVM_CHAIN_IDS_WITH_TRANSACTIONS:
+                        if len(location_labels := cursor.execute(  # number of earliest receive events of distinct assets after last_save_time  # noqa: E501
+                            'SELECT DISTINCT location_label FROM '
+                            '(SELECT MIN(timestamp), location_label FROM history_events '
+                            f'WHERE (type IN ({",".join(["?"] * len(type_filter))}) OR '
+                            f'subtype IN ({",".join(["?"] * len(subtype_filter))})) '
+                            'AND location = ? AND timestamp >= ? GROUP BY asset);',
+                            type_filter + subtype_filter + [
+                                Location.from_chain_id(chain).serialize_for_db(),
+                                last_save_time * 1000,  # to ms
+                            ],
+                        ).fetchall()) > 0:
+                            detect_tokens[chain.to_blockchain()] = [
+                                string_to_evm_address(location_label[0])
+                                for location_label in location_labels
+                            ]
```

Comment:
am I missing something or you didn't use the variable in the task below?

should be something like

```
detect_tokens=detect_tokens if len(detec_tokens) else None